packages/verify/lib/verify.ts

```
@@ -1,120 +1,91 @@
 import { Client } from '@vonage/server-client';
-import { Command } from './enums/Command';
-import { PSD2Request } from './interfaces/PSD2Request';
-import { VerifyCheckResponse } from './interfaces/Response/VerifyCheckResponse';
-import { VerifyControlResponse } from './interfaces/Response/VerifyControlResponse';
-import { VerifySearchResponse } from './interfaces/Response/VerifySearchResponse';
-import { VerificationRequest } from './interfaces/VerificationRequest';
-
-const remapObjects = <T, O>(mapping, newObject: T, oldObject: O): T => {
-  for (const key in mapping) {
-    if (oldObject[mapping[key]]) {
-      newObject[key] = oldObject[mapping[key]];
-      delete oldObject[mapping[key]];
-    }
-  }
-  newObject = { ...newObject, ...oldObject };
-  return newObject;
-};
+import { Command } from './enums/index';
+import {
+  VerifyControl,
+  VerifyControlError,
+  VerifyControlErrorResponse,
+  VerifyControlResponse,
+  VerifyCheckError,
+  VerifyCheck,
+  VerifyCheckResponse,
+  VerifyRequestResponse,
+  VerifySearchResponse,
+  VerifySearch,
+  VerifySearchError,
+  VerifyRequestErrorResponse,
+  VerifyError,
+  VerifyRequest,
+} from './interfaces/index';
+import { VerificationParameters, PSD2Parameters } from './types/index';
 
 export class Verify extends Client {
-  public async cancel(requestId: string): Promise<VerifyControlResponse> {
+  public async sendControl(
+    command: Command,
+    requestId: string,
+  ): Promise<VerifyControl | VerifyControlError> {
     const data = {
       request_id: requestId,
       cmd: Command.CANCEL,
     };
 
-    const resp = await this.sendPostRequest<VerifyControlResponse>(
-      `${this.config.apiHost}/verify/control/json`,
-      data,
-    );
-    return resp.data;
+    // eslint-disable-next-line max-len
+    const resp = await this.sendPostRequest<
```

This is probably going to be in a few different places, but for consistency I think we should actually throw an Error instead of returning an error. This is more consistent with the newer APIs work, and much easier to find and handle than returning an error.

That would be a BC break, so I would document that behavior change, or make it forward compatible with an options flag that will allow you to keep existing behavior or throw an error (thereby deprecating the old behavior and allowing the user a way to opt-in to new behavior to make upgrading easier). 

Just because the API is bad and doesn't return proper HTTP codes doesn't mean the SDK shouldn't behave as expected.

---

I want to avoid having this PR being a BC one. This was trying to address issues with the types. When we start working on verify V2, then we'll go back and have this throw an exception.